azurerm/_modules/aks/main.tf

```diff
@@ -47,3 +60,14 @@ resource "azurerm_kubernetes_cluster" "current" {
 
   tags = var.metadata_labels
 }
+
+data "external" "aks_nsg_id" {
+  program = [
+    "bash",
+    "${path.root}/scripts/aks_nsg_name.sh"
```

Comment:
You're currently placing the script in the quickstart. This means it would be part of the user repository. The GKE implementation requires a similar script. By placing the script inside the module, it will be updated to match the module's version. Leaving it in the quickstart, would require users to update the module and the script individually.

Here's the GKE example: https://github.com/kbst/terraform-kubestack/blob/master/google/_modules/gke/cluster_role_binding.tf#L2